src/models.ts

```diff
@@ -92,4 +92,31 @@ export const createJSONRPCErrorResponse = (
   };
 };
 
+export const createJSONRPCSuccessResponse = (
+  id: JSONRPCID,
+  result?: any
+): JSONRPCSuccessResponse => {
+  return {
+    jsonrpc: JSONRPC,
+    id,
+    result: result ?? null,
+  };
+};
+
+export const createJSONRPCRequest = (
+  method: string,
+  params?: JSONRPCParams,
+  id?: JSONRPCID
+): JSONRPCRequest => {
+  const response: JSONRPCRequest = {
+    jsonrpc: JSONRPC,
+    method,
+    params,
+  };
+
+  if (id !== undefined) response.id = id;
+
+  return response;
+};
```

Comment:
Can you please separate a factory function for notifications?

Other functions have `id` as the first parameter, and in most of the common use case, we want `id`. I'm concerned having it as an optional parameter at the end makes it inconsistent with other functions and potentially makes it prone to forget passing `id`.
```suggestion
export const createJSONRPCRequest = (
  id: JSONRPCID,
  method: string,
  params?: JSONRPCParams
): JSONRPCRequest => {
  return {
    jsonrpc: JSONRPC,
    id,
    method,
    params,
  };
};

export const createJSONRPCNotification = (
  method: string,
  params?: JSONRPCParams
): JSONRPCRequest => {
  return {
    jsonrpc: JSONRPC,
    method,
    params,
  };
};
```